scripts/controllers/main.js

```diff
@@ -1274,12 +1261,57 @@ App.controller('Main', function ($scope, $timeout, $location, Api, tmhDynamicLoc
       return false;
    };
 
+   $scope.climateTarget = function (item, entity) {
+      const value = entity.attributes.temperature || [
+         entity.attributes.target_temp_low,
+         entity.attributes.target_temp_high,
+      ].join(' - ');
+
+      if (item.filter) {
+         return item.filter(value);
+      }
+
+      return value;
+   };
+
+   $scope.reverseLookupClimateOption = function (item, entity, option) {
+      initializeClimateOptions(item, entity);
+      for (const [key, value] of Object.entries(item.climateOptions)) {
+         if (value === option) {
+            return key;
+         }
+      }
+      return option;
+   };
+
+   $scope.lookupClimateOption = function (item, entity, option) {
+      initializeClimateOptions(item, entity);
+      return item.climateOptions[option] || option;
+   };
+
+   function initializeClimateOptions (item, entity) {
+      if (typeof item.climateOptions === 'undefined') {
+         const _options = item.useHvacMode ? entity.attributes.hvac_modes : entity.attributes.preset_modes;
+         const options = {};
+         for (const option of _options) {
+            if (item.states !== null && typeof item.states === 'object') {
+               options[option] = item.states[option] || option;
+            } else {
+               options[option] = option;
+            }
+         }
+         item.climateOptions = options;
+      }
+   }
```

Comment:
Here is my suggestion that reuses `item.states`, merging auto-discovered and user values together if the user did not include some. Also, this will make `item.states` the source of truth instead of creating a separate property for essentially the same thing.

```suggestion
   function initializeClimateStates (item, entity) {
      if (item.__statesExtended === undefined) {
         item.states = item.states || {};
         const options = item.useHvacMode ? entity.attributes.hvac_modes : entity.attributes.preset_modes;
         for (const option of options) {
            if (!(option in item.states)) {
               item.states[option] = option;
            }
         }
         item.__statesExtended = true;
      }
   }
```

Note that I've also renamed the function for clarity.

Reply:
item.__statesExtended are not only states, they can be also preset_modes that does not belong to entity state... it is an attribute

Reply:
so the correct name options...

Reply:
I feel like my suggested naming still works, even if the `preset_modes` are not really entity states. I mean if the user wants to localize `preset_modes` then he has to do that in `item.states` right? So from the user's perspective, it does appear to be "states" still.

(btw. the localization of states could be documented also since it's not entirely obvious how it works, especially with the `hvac_modes` and `preset_modes` being relevant here.